src/graph/transform/cuda/knn.cu

```diff
@@ -307,15 +307,18 @@ __global__ void BruteforceKnnShareKernel(
   FloatType* data_buff = SharedMemory<FloatType>();
   FloatType* query_buff = data_buff + block_size * feature_size;
   FloatType* dist_buff = query_buff + block_size * feature_size;
-  IdType* res_buff = reinterpret_cast<IdType*>(dist_buff + block_size * k);
+  IdType* res_buff = reinterpret_cast<IdType*>(
```

Comment:
should we check the size before `reinterpret_cast` to avoid unexpected access breach? `static_assert(sizeof(FloatType) >= sizeof(IdType), "Not allowed to reinterprect_cast...")` ?

Reply:
Fixed with aligned shared memory memory access.